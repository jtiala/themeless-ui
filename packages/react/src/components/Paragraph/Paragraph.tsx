import { CommonComponentProps, cn } from "@themeless-ui/utils";
import { ReactNode } from "react";

export type ParagraphProps = {
  /**
   * Paragraph content.
   */
  children?: ReactNode;
} & CommonComponentProps;

const className = cn("paragraph");

/**
 * A paragraph of text.
 */
export function Paragraph({ children, id, testId }: ParagraphProps) {
  return (
    <p className={className} id={id} data-testid={testId}>
      {children}
    </p>
  );
}
