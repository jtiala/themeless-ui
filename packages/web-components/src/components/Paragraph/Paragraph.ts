import style from "@themeless-ui/style/dist/paragraph.css?inline";
import { cn } from "@themeless-ui/utils";
import { html, unsafeCSS } from "lit";
import { customElement } from "lit/decorators.js";
import { TUIComponent } from "../../utils";

const className = cn("paragraph");

@customElement("tui-paragraph")
export class Paragraph extends TUIComponent {
  static styles = unsafeCSS(style);

  override render() {
    return html`
      <p class="${className}" id="${this.id}" data-testid="${this.testId}">
        <slot></slot>
      </p>
    `;
  }
}

declare global {
  interface HTMLElementTagNameMap {
    "tui-paragraph": Paragraph;
  }
}
