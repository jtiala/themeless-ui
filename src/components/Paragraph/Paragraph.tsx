import { ReactNode } from "react";
import { cn } from "../../utils/css";

export type ParagraphProps = {
  /**
   * Paragraph content.
   */
  children?: ReactNode;
};

const className = cn("paragraph");

/**
 * Display a text paragraph.
 */
export function Paragraph({ children }: ParagraphProps) {
  return <p className={className}>{children}</p>;
}
