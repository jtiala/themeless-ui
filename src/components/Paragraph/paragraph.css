.tui__paragraph {
  display: block;
  margin: 0 0 var(--tui__spacing-8) 0;
  color: var(--tui__color-black);
}
