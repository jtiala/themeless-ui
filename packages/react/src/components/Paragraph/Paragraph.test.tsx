import {
  componentToJson,
  describe,
  expect,
  it,
  render,
  renderer,
  screen,
} from "../../../test";
import { Paragraph } from "./Paragraph";

const element = (
  <Paragraph testId="paragraph">Lorem ipsum dolor sit amet.</Paragraph>
);

describe("Paragraph", async () => {
  it("should render the paragraph", () => {
    expect(componentToJson(renderer.create(element))).toMatchSnapshot();

    render(element);

    const paragraph = screen.getByTestId("paragraph");

    expect(paragraph).toBeInTheDocument();
    expect(paragraph).toHaveTextContent("Lorem ipsum dolor sit amet.");
  });
});
