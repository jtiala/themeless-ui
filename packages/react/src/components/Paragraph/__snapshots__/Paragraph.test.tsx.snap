// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Paragraph > should render the paragraph 1`] = `
<p
  className="tui__paragraph"
  data-testid="paragraph"
>
  Lorem ipsum dolor sit amet.
</p>
`;
